src/Storages/MergeTree/MergeTreeIndexGranularity.cpp

```diff
@@ -120,6 +120,19 @@ size_t MergeTreeIndexGranularity::countMarksForRows(size_t from_mark, size_t num
     return getRowsCountInRange(from_mark, std::max(1UL, to_mark)) - offset_in_rows;
 }
 
+size_t MergeTreeIndexGranularity::countMarksForRows(size_t from_mark, size_t number_of_rows) const
```

Comment:
I understand the semantics of this function and why it is called in the same way. But I do not understand why another function is called the same: countMarksForRows. It counts number of rows, not marks as far sa I understand. Also almost all the code is duplicated (except the return statement). Why do we need both of them with the same name?